### PR TITLE
Add more default extensions

### PR DESCRIPTION
**Pull request by phadej**
... and cleanup few headers

### CODE REVIEW

File: lsm-tree.cabal [hunk 1]
```diff
@@ -34,7 +34,14 @@ common wno-x-partial
 common language
   default-language:   GHC2021
   default-extensions:
+    DeriveAnyClass
+    DerivingStrategies
+    DerivingVia
+    ExplicitNamespaces
     GADTs
+    LambdaCase
+    MultiWayIf
```
Comment (dcoutts) [inline lsm-tree.cabal:43]: I'm not especially keen on encouraging this one. It looks like it's used in only one place at the moment.

File: lsm-tree.cabal [hunk 2]
```diff
@@ -34,7 +34,14 @@ common wno-x-partial
 common language
   default-language:   GHC2021
   default-extensions:
+    DeriveAnyClass
+    DerivingStrategies
+    DerivingVia
+    ExplicitNamespaces
```
Comment (dcoutts) [inline lsm-tree.cabal:40]: This one doesn't seem to be used anywhere.

File: lsm-tree.cabal [hunk 3]
```diff
@@ -34,7 +34,14 @@ common wno-x-partial
 common language
   default-language:   GHC2021
   default-extensions:
+    DeriveAnyClass
```
Comment (dcoutts) [inline lsm-tree.cabal:37]: This one conflicts with `GenericNewtypeDeriving`. See the build failures.

**Review by dcoutts — COMMENTED** [review]
I agree with the general direction.

File: lsm-tree.cabal (hunk 2 above)
Comment (phadej) [reply lsm-tree.cabal:40]: See docs for https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/control.html#extension-GHC2021

`ExplicitNamespaces` is transitively enabled by `TypeOperators`, but not by `GHC2021`.

File: lsm-tree.cabal (hunk 1 above)
Comment (phadej) [reply lsm-tree.cabal:43]: It's either one line here or there. Either we use it, or not use it at all.

File: lsm-tree.cabal (hunk 3 above)
Comment (phadej) [reply lsm-tree.cabal:37]: Arguably we could enable `-Wmissing-deriving-strategies` (with `Werror`).

File: lsm-tree.cabal (hunk 2 above)
Comment (dcoutts) [reply lsm-tree.cabal:40]: I see thanks.

File: lsm-tree.cabal (hunk 1 above)
Comment (dcoutts) [reply lsm-tree.cabal]: You're quite right. I've added a patch.

**Review by dcoutts — APPROVED** [review]

**Pull request merged**